interfaces/builtin/basedeclaration.go

```diff
@@ -178,6 +178,12 @@ slots:
         - core
         - gadget
     deny-auto-connection: true
+  boot-config:
+    allow-installation:
+      slot-snap-type:
+        - gadget
+    deny-connection: true
+    deny-auto-connection: true
```

Comment:
Either sounds ok to me, maybe we can talk online to see which is the best option.